lib/rucio/core/transfer.py

```diff
@@ -189,19 +189,37 @@ def __rewrite_source_url(source_url, source_sign_url, dest_sign_url, source_sche
         return source_url
 
     @staticmethod
-    def __rewrite_dest_url(dest_url, dest_sign_url):
+    def __rewrite_dest_url(dest_url: str, rewrite_url: str, values: "Optional[Union[Dict, None]]" = None) -> str:
         """
-        Parametrize destination url for some special cases of destination schemes
+        Parametrize destination url for some special cases
         """
-        if dest_sign_url == 'gcs':
+
+        # FTS specific protocol translations for cloud storage
+        if rewrite_url == 'gcs':
             dest_url = re.sub('davs', 'gclouds', dest_url)
             dest_url = re.sub('https', 'gclouds', dest_url)
-        elif dest_sign_url == 's3':
+        elif rewrite_url == 's3':
             dest_url = re.sub('davs', 's3s', dest_url)
             dest_url = re.sub('https', 's3s', dest_url)
 
+        # Archival colocation hints
+        if rewrite_url == 'archive_metadata' and values is not None:
+            # Consider already existing URL parameters
+            if '?' in dest_url:
```

Comment:
Wouldn't it be easier to use the `add_url_query` function from common/utils to add the query parameters ?